mava/systems/tf/maddpg/execution.py

```diff
@@ -147,12 +147,19 @@ def select_action(
         # Step the recurrent policy/value network forward
         # given the current observation and state.
         action, policy = self._policy(agent, observation.observation)
-
-        # Return a numpy array with squeezed out batch dimension.
-        action = tf2_utils.to_numpy_squeeze(action)
-        policy = tf2_utils.to_numpy_squeeze(policy)
         return action, policy
 
+    @tf.function
```

Comment:
@DriesSmit @RuanJohn we should just confirm this change didn't cause the memory leak issue.

Reply:
Discussed offline. Still under investigation but is not particular to MADDPG.